packages/revocation/src/implementations/revocationOnChain.ts

```diff
@@ -0,0 +1,127 @@
+import {
+  Contract, ethers, Event, utils,
+} from 'ethers';
+import { EwSigner, addressOf } from '@ew-did-registry/did-ethr-resolver';
+import { abi as RevocationOnChainAbi } from '../constants/RevocationRegistryOnChain.json';
+
+export class RevocationOnChain {
+  private _revocationRegistryOnChain: Contract;
+
+  /**
+  * @param owner - Entity which controls revocation
+  * @param addressOnchain - Address of the on chain claim's RevocationRegistry
+  */
+  constructor(
+    owner: EwSigner,
+    addressOnChain: string,
+  ) {
+    this._revocationRegistryOnChain = new ethers.Contract(
```

Comment:
Not absolutely necessary, but it could be nice to compile a Contract factory type using typechain to use here (see iam-client-lib package.json for example). What do you think of this?

Reply:
Yeah, sounds good to me. Will do so.

Reply:
This change could be made at a later stage as the current version of `TS` in the `ew-did-registry` is `3.9.7` and it does create error when using `@typechain`. Once the `TS` version is updated, we would add the factory build for contracts.